<?xml version="1.0" encoding="utf-8"?>

<launch>
  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video2" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="color_format" value="yuv422p" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
  </node>

  <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
    <remap from="image" to="/usb_cam/image_raw"/>
    <param name="autosize" value="true" />
  </node>

  <!--启动yolo_v5 启动权重-->
  <param name="yolov5_path"       value="$(find yolov5_ros)/yolov5"/>
  <param name="use_cpu"           value="false" />
    <node pkg="yolov5_ros" type="yolo5v.py" name="yolov5_ros" output="screen" >
    <param name="weight_pathK"       value="$(find yolov5_ros)/weights/weightK/best.pt"/>
    <param name="weight_pathR"       value="$(find yolov5_ros)/weights/weightR/best.pt"/>
    <param name="weight_pathB"       value="$(find yolov5_ros)/weights/weightB/best.pt"/>
    <param name="weight_path1"       value="$(find yolov5_ros)/weights/weight1/best.pt"/>
    <param name="image_topic"       value="/usb_cam/image_raw" />
    <param name="pub_topicK"         value="pub_topicK" />
    <param name="pub_topicR"         value="pub_topicR" />
    <param name="pub_topicB"         value="pub_topicB" />
    <param name="pub_topicRB"         value="pub_topicRB" />
    <param name="camera_frame"       value="camera_color_frame"/>
    <param name="conf"              value="0.6" />
    <param name="iou"               value="0.8" /> 
  </node>


  <node pkg="yolov5_test" type="cam_5" name="cam_5" output="screen"/>

    
  <arg name="port_name" default="/dev/ttyUSB0" />
  <arg name="baud_value" default="115200" />
  <arg name="serial_node" default="serial_node" />
    <node pkg="rosserial_python" type="serial_node.py" name="$(arg serial_node)" output="screen">
        <param name="port" value="$(arg port_name)" />
        <param name="baud" value="$(arg baud_value)" />
  </node>
</launch>
    


